packages/plugins/src/plugins/curve/depositFetcher.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Address,
+  ContractFunctionConfig,
+  MulticallResult,
+  getAddress,
+} from 'viem';
+import { Cache } from '../../Cache';
+import { platformId, vestingSplitterContracts } from './constants';
+import { Fetcher, FetcherExecutor } from '../../Fetcher';
+import { getEvmClient } from '../../utils/clients';
+import { ElementRegistry } from '../../utils/elementbuilder/ElementRegistry';
+import { zeroBigInt } from '../../utils/misc/constants';
+import { vestingSplitterABI } from './abis';
+import { MAINNET_NETWORK_ID } from '../../utils/evm/constants';
+
+const executor: FetcherExecutor = async (owner: string, cache: Cache) => {
+  const ownerAddress = getAddress(owner);
+  const client = getEvmClient(MAINNET_NETWORK_ID);
```

Comment:
We should wrap this in a factory like we've done in our most recent PRs to avoid hard-coding for a specific network.

```suggestion
const executor: FetcherExecutor = async (owner: string, cache: Cache) => {
  const ownerAddress = getAddress(owner);
  const client = getEvmClient(networkId);
```

Reply:
See https://github.com/Octav-Labs/sonarwatch-portfolio/blob/86fa8e2cf8abd89702a0cf79bc08eb76eec66cac/packages/plugins/src/plugins/metronome/metronomeFetcher.ts#L15-L21 as a reference.